operator/watchers/cilium_podippool.go

```diff
@@ -27,6 +137,10 @@ func StartIPPoolAllocator(
 ) {
 	log.Info("Starting CiliumPodIPPool allocator watcher")
 
+	if len(operatorOption.Config.IPAMAutoCreateCiliumPodIPPools) > 0 {
```

Comment:
I think this check is redundant because `multiPoolAutoCreatePools` will range over the map at the top level, i.e. it's a no-op in case of an empty map. Might still be good to keep an explicit check in case we change ever `multiPoolAutoCreatePools` to perform other operations.